### PR TITLE
[MWAR-444] Update plugin (requires Maven 3.2.5+)

### PR DESCRIPTION
**Pull request by cstamas**
Move maven to proper scope, update dependencies,
remove unused ones.

---

https://issues.apache.org/jira/browse/MWAR-444

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -62,9 +62,9 @@
   </distributionManagement>
 
   <properties>
-    <mavenArchiverVersion>3.5.0</mavenArchiverVersion>
+    <mavenArchiverVersion>3.5.1</mavenArchiverVersion>
     <mavenFilteringVersion>3.1.1</mavenFilteringVersion>
-    <mavenVersion>3.1.0</mavenVersion>
+    <mavenVersion>3.2.5</mavenVersion>
```
[the remark this answers is not in the document]
Comment (slawekjaranowski) [reply pom.xml:67]: `brew` on Mac OS install the latest version of Maven

**Review by slawekjaranowski — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -129,7 +133,7 @@
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-archiver</artifactId>
-      <version>4.2.2</version>
+      <version>4.2.5</version>
```
Comment (michael-o) [inline pom.xml:136]: 4.2.7 is out

Comment (jorsol) [reply pom.xml:136]: > 4.2.7 is out

But until the maven-war-plugin uses Java 8, that would require to downgrade commons-compress and plexus-utils, and my guess is that to simplify transitive dependency management, this just sets the latest that has bytecode 51.

Comment (slachiewicz) [reply pom.xml:136]: We are moving everything to 8

Comment (jorsol) [reply pom.xml:136]: Sure, yet this project is still Java 7:
https://github.com/apache/maven-war-plugin/blob/4355af5a697de7c1a782da8aed6f0b45993bf384/pom.xml#L68

So, ideally bump here to Java 8 and update plexus-archiver to 4.2.7, or bump it on another PR.

Comment (slachiewicz) [reply pom.xml:136]: PR opened https://github.com/apache/maven-war-plugin/pull/23

**Comment by jira-importer** [comment]
Resolve #516

**Pull request merged**